src/containers/MuppetDetailPage.test.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import MuppetDetailPage from './MuppetDetailPage';
+import { getMuppetById } from '../services/muppetApi';
+
+jest.mock('../services/muppetApi.js');
+
+describe('MuppetDetailPage container', () => {
+  it('displays muppet details', async() => {
+    getMuppetById.mockResolvedValue(
+      { name: 'Kermit', performer: 'Jim Henson', image: 'test.png' }
+    );
+
+    render(<MuppetDetailPage
+      match={{
+        params: { muppetName: 'Kermit' }
+      }}
```

Comment:
instead of passing a match prop you should wrap the component in a memoryrouter and mimic the actual behavior of going to the proper route